src/cljs/harja/ui/grid/perus.cljs

```diff
@@ -360,14 +360,19 @@
                                               (when (raporttivienti :pdf)
                                                 pdf-nappi))
               aseta-parametrit! (fn [id]
-                                  (let [input (-> js/document
-                                                  (.getElementById id)
-                                                  (aget "parametrit"))
-                                        parametrit raporttiparametrit]
-                                    (set! (.-value input)
-                                          (tr/clj->transit parametrit))
-                                    true))
-              
+                                  ;; Meillä voi olla tallenna- nappeja ylhäällä sekä alhaalla eli yhteensä 4 kpl
+                                  ;; Tässä käy niin että tulee 2 paria nappia samoilla IDllä, täytyy antaa kaikille napeille raporttiparametrit
```

Comment:
Mielestäni pitäisi mieluummin korjata tämä, että meillä on useita nappeja samalla ID:llä.

Reply:
Tämä ei käytännössä mikään ongelma ole vaikka niitä nappeja on samalla ID:llä, mutta tutkiskelin asiaa niin kyllä siitä vain voi ongelmia tulla selainten kanssa koska tämä on kuulemma invalid HTML. 

Jätin silti tuon koodin joka looppaa kaikki olemassa olevat IDt läpi "failsafeksi", vaikka niitä IDitä onkin nyt vain 1 kpl.
Alemmalle "muokkauspaneelille" annoin uuden parametrin muokkauspaneeli-alhaalla? true, ja sen perusteella passataan omat IDt napeille funktioon **_(defn tallenna-excel-nappi)_**.

https://github.com/finnishtransportagency/harja/pull/2912/commits/0693931f5813c698bde1c7b9899a8e2acaf58376

Reply:
Juuri näin, duplikaatti-IDt toimivat kyllä käytännössä kaikissa selaimissa, mutta ne silti ovat virheellistä HTML:lää, eikä niitä saisi hyväksyä. Tämä muutos oikein hyvä :+1: